### ISSUE TITLE
Support Google CloudSQL Second Generation instance type

### ISSUE THREAD

**Issue by evandbrown** [hashicorp/terraform#8943]
Second Generation instance types became generally available in August 16 (https://cloudplatform.googleblog.com/2016/08/Cloud-SQL-Second-Generation-performance-and-feature-deep-dive.html). Modify the Cloud SQL provider to support this instance type.

**Comment by csainty** [comment]
I'd love to see this included.
In particular I want to be able to set...

```
dataDiskSizeGb: 100
dataDiskType: PD_SSD
storageAutoResize: true
failoverReplica:
    name: xxx
maintenanceWindow:
    day: 2
    hour: 10
```

Do you need external contributions around it or is there work underway to make these updates?

**Comment by marcjoha** [comment]
Basic support for 2nd gen instances is already available. The key thing to notice is that 'tier' should specify the instance type if 2nd gen instances are created, and the pricing tier if 1st gen instances are created. This is mentioned here: https://cloud.google.com/sql/docs/admin-api/v1beta4/tiers.

Here's an example creating a 2nd gen instance. Also notice that the region parameter is different depending which generation is used.

resource "google_sql_database_instance" "master" {
  name = "master-instance"
  region = "us-central1"
  database_version = "MYSQL_5_7"
  settings = {
    tier = "db-n1-standard-1"
  }
}

**Comment by stack72** [comment]
Closed via #11571
